src/tests/class_tests/openms/source/FragmentMassError_test.cpp

```diff
@@ -0,0 +1,291 @@
+
+  PeptideIdentification peptide_id;
+  peptide_id.setMetaValue("spectrum_reference",id);
+  peptide_id.setMZ(mz);
+  peptide_id.setHits({peptide_hit});
+
+  return peptide_id;
+}
+
+START_TEST(FragmentMassError, "$Id$")
+
+  FragmentMassError* ptr = nullptr;
+  FragmentMassError* nulpt = nullptr;
+  START_SECTION(FragmentMassError())
+  {
+    ptr = new FragmentMassError();
+    TEST_NOT_EQUAL(ptr, nulpt)
+  }
+  END_SECTION
+
+  START_SECTION(~FragmentMassError())
+  {
+    delete ptr;
+  }
+  END_SECTION
+
+
+  FragmentMassError frag_ma_err;
+
+  //tests compute function
+  START_SECTION(void compute(FeatureMap& fmap, const MSExperiment& exp, const std::map<String,UInt64>& map_to_spectrum, const ToleranceUnit tolerance_unit = ToleranceUnit::AUTO, const double tolerance = 20))
+  {
+    //--------------------------------------------------------------------
+    // create valid input data
+    //--------------------------------------------------------------------
+    // FeatureMap
+    FeatureMap fmap;
+
+    // empty PeptideIdentification
+    PeptideIdentification pep_id_empty;
+    pep_id_empty.setRT(6);
+
+    // empty Feature
+    Feature feat_empty;
+
+    // put valid data in fmap
+    fmap.setUnassignedPeptideIdentifications({createPeptideIdentification("XTandem::1", "HIMALAYA", 1, 888), createPeptideIdentification("XTandem::2", "ALABAMA", 2, 264), pep_id_empty});
+    fmap.push_back(feat_empty);
+    // set ProteinIdentifications
+    ProteinIdentification protId;
+    ProteinIdentification::SearchParameters param;
+    param.fragment_mass_tolerance_ppm = false;
+    param.fragment_mass_tolerance = 0.3;
+    protId.setSearchParameters(param);
+    fmap.setProteinIdentifications( {protId} );
+
+    // MSExperiment
+    MSExperiment exp;
+
+    // create b- and y-ion spectrum of peptide sequence HIMALAYA with charge 1
+    // shift every peak by 0.001 mz
+    PeakSpectrum ms_spec_2_himalaya = createMSSpectrum(2, 3.7, "XTandem::1");
+    TheoreticalSpectrumGenerator theo_gen_hi;
+    theo_gen_hi.getSpectrum(ms_spec_2_himalaya, AASequence::fromString("HIMALAYA"), 1, 1);
+    for(Peak1D& peak : ms_spec_2_himalaya) peak.setMZ(peak.getMZ() + 0.001);
+
+    // create c- and z-ion spectrum of peptide sequence ALABAMA with charge 2
+    // shift every peak by 0.001 mz
+    PeakSpectrum ms_spec_2_alabama = createMSSpectrum(2, 2, "XTandem::2", Precursor::ActivationMethod::ECD);
+    TheoreticalSpectrumGenerator theo_gen_al;
+    Param theo_gen_settings_al = theo_gen_al.getParameters();
+    theo_gen_settings_al.setValue("add_c_ions", "true");
+    theo_gen_settings_al.setValue("add_z_ions", "true");
+    theo_gen_settings_al.setValue("add_b_ions", "false");
+    theo_gen_settings_al.setValue("add_y_ions", "false");
+    theo_gen_al.setParameters(theo_gen_settings_al);
+    theo_gen_al.getSpectrum(ms_spec_2_alabama, AASequence::fromString("ALABAMA"), 2, 2);
+    for(Peak1D& peak : ms_spec_2_alabama) peak.setMZ(peak.getMZ() + 0.001);
```

Comment:
I think it would be easier to verify tests if instead of 0.001 e.g. ground_truth_ppm * peak.getMZ() was used.